components/sign/ui-sign-input/SignPasswordInput.tsx

```diff
@@ -0,0 +1,89 @@
+import styles from "./SignInput.module.scss";
+import classNames from "classnames/bind";
+import { SignPasswordInputProps, isPasswordValid } from "./constant";
+import Image from "next/image";
+import { useState } from "react";
+import { useRouter } from "next/router";
+
+const cx = classNames.bind(styles);
+
+export const SignPasswordInput = ({
+  values,
+  setValues,
+  passwordErrorMessageRef,
+  passwordErrorInputRef,
+  showPasswordError,
+  setShowPasswordError,
+}: SignPasswordInputProps) => {
+  const router = useRouter();
+  const path = router.pathname;
+
+  const [showPassword, setShowPassword] = useState(false);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValues({ ...values, password: e.target.value });
+  };
+
+  const handleBlur = () => {
+    setShowPasswordError(true);
+
+    if (values.password === "") {
+      passwordErrorMessageRef.current.textContent = "비밀번호를 입력해주세요.";
+      return;
+    }
+
+    if (path === "/signup" && !isPasswordValid(values.password)) {
+      passwordErrorMessageRef.current.textContent =
+        "비밀번호는 영문, 숫자 조합 8자 이상 입력해 주세요.";
+      return;
+    }
+
+    if (values.password) {
+      return setShowPasswordError(false);
+    }
+  };
```

Comment:
```suggestion
const handleBlur = () => {
  // 에러 메시지 초기화
  let errorMessage = "";

  if (values.password === "") {
    // 비밀번호 입력되지 않은 경우
    errorMessage = "비밀번호를 입력해주세요.";
  } else if (path === "/signup" && !isPasswordValid(values.password)) {
    // 회원가입 경로에서 비밀번호 유효성 검사 실패
    errorMessage = "비밀번호는 영문, 숫자 조합 8자 이상 입력해 주세요.";
  }

  // 에러 메시지 업데이트
  if (errorMessage) {
    passwordErrorMessageRef.current.textContent = errorMessage;
    setShowPasswordError(true);
  } else {
    setShowPasswordError(false);
  }
};

```

여기도 이렇게 수정할 수 있겠네여

마지막 조건식은 값이 있어서 if문이 불필요하기때문에 에러메시지를 조건으로 판별할 수 있겠습니다.